packages/framework-core/test/services/graphql/graphql-generator.test.ts

```diff
@@ -1,38 +1,53 @@
-import { SinonStubbedInstance, replace, restore, SinonStub, stub } from 'sinon'
+import { replace, restore, SinonStub, stub } from 'sinon'
 import { BoosterCommandDispatcher } from '../../../src/booster-command-dispatcher'
-import sinon = require('sinon')
 import { BoosterReadModelDispatcher } from '../../../src/booster-read-model-dispatcher'
 import { GraphQLGenerator } from '../../../src/services/graphql/graphql-generator'
-import { BoosterConfig, ReadModelInterface } from '@boostercloud/framework-types/dist'
+import { BoosterConfig, ReadModelInterface, Logger, Level } from '@boostercloud/framework-types/dist'
 import { expect } from '../../expect'
 import { GraphQLQueryGenerator } from '../../../src/services/graphql/graphql-query-generator'
 import { GraphQLMutationGenerator } from '../../../src/services/graphql/graphql-mutation-generator'
 import { GraphQLSubscriptionGenerator } from '../../../src/services/graphql/graphql-subcriptions-generator'
 import { random, internet, lorem } from 'faker'
+import { buildLogger } from '../../../src/booster-logger';
 
 describe('GraphQL generator', () => {
   let mockEnvironmentName: string
   let mockConfig: BoosterConfig
-
-  let commandDispatcherStub: SinonStubbedInstance<BoosterCommandDispatcher>
-  let readModelDispatcherStub: SinonStubbedInstance<BoosterReadModelDispatcher>
-
+  let mockLogger: Logger
   let sut: GraphQLGenerator
 
   beforeEach(() => {
     mockEnvironmentName = random.alphaNumeric(10)
     mockConfig = new BoosterConfig(mockEnvironmentName)
-
-    commandDispatcherStub = sinon.createStubInstance(BoosterCommandDispatcher)
-    readModelDispatcherStub = sinon.createStubInstance(BoosterReadModelDispatcher)
-
-    sut = new GraphQLGenerator(mockConfig, commandDispatcherStub, readModelDispatcherStub as any)
+    mockLogger = buildLogger(Level.error)
+    sut = GraphQLGenerator.getInstance(mockConfig, mockLogger)
   })
 
   afterEach(() => {
     restore()
   })
 
+  describe('getInstance', () => {
+    
+    it('should create the instance', () => {
+      const instance = GraphQLGenerator.getInstance(mockConfig, mockLogger)
+
+      expect(instance).to.not.be.undefined
+      expect(instance).to.be.an.instanceOf(GraphQLGenerator)
+    })
+
+    it('should return always the same instance', () => {
+      const instance1 = GraphQLGenerator.getInstance(mockConfig, mockLogger)
+      const instance2 = GraphQLGenerator.getInstance(mockConfig, mockLogger)
```

Comment:
Given that neither the config or the logger ever change, maybe we should explore a way to make these params disappear. I think that the config should definitely become a singleton, and the logger can be a config option (not in this PR anyway).

One nice advantage of this is that we could allow the user to set an alternate logger class in their Booster config to easily support logging to external platforms like Splunk or Papertrail...

Reply:
You are right, we can add more refactoring here. I'll create the issue for future improvements

Reply:
@javiertoledo what you describe is addressed with the [Reader pattern](https://dev.to/riccardo_cardin/and-monads-for-almost-all-the-reader-monad-1ife), sadly TypeScript syntax doesn't support working comfortably with it. I'll _eventually_ work on a fix...

Reply:
I have created an issue for this future refactoring https://github.com/boostercloud/booster/issues/566